pkgs/build-support/setup-hooks/wrap-gapps-hook/tests/lib.nix

```
@@ -19,12 +20,14 @@ rec {
   '';
 
   expectSomeLineContainingYInFileXToMentionZ = file: filter: expected: ''
-    if ! cat "${file}" | grep "${filter}"; then
-        ${fail "The file “${file}” should include a line containing “${filter}”."}
+    file=${escapeShellArg file} filter=${escapeShellArg filter} expected=${escapeShellArg expected}
```

```suggestion
    file=${lib.escapeShellArg file} filter=${lib.escapeShellArg filter} expected=${lib.escapeShellArg expected}
```

`with` should not be used over the entire file.

---

Is there prior discussion on this? I feel like `with lib` and `with builtins` are acceptable.

---

GitHubs search is trash so I will never find it but `with something` over the entire file is a big performance impact especially on large files like modules. Best to not introduce more usages of it or keep them small.

---

Oh ok, that's weird